flag.go

```diff
@@ -896,6 +899,25 @@ func (f *FlagSet) usage() {
 	}
 }
 
+//--unknown (args will be empty)
+//--unknown --next-flag ... (args will be --next-flag ...)
```

Comment:
What if I passed `--unknown=bob --next-flag`. I think this will strip `--next-flag`, but I don't think it should.

Reply:
this function just strips value for unknown flag, the flag itself is not included in args. 

for `--unknown=value --next-flag ...`   args will be --next-flag ...

added the testcase to test this explicitly

Reply:
I really think you need to know `len(split)` (from `parseLongArg`) and to use that. What if it was `--unknown=value subcmd`.  You would strip `subcmd` which is really not what the user wants.

Clearly it is a heuristic to know what to strip and what not to strip for an 'unknown' but I think we want to be as careful as we can...

Reply:
oh yes, thats right. i would consider this a bug :)

fixed. thank you for pointing out.